csharp/ql/lib/semmle/code/csharp/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -21,7 +21,13 @@ private import semmle.code.csharp.frameworks.system.collections.Generic
  */
 DotNet::Callable getCallableForDataFlow(DotNet::Callable c) {
   exists(DotNet::Callable unboundDecl | unboundDecl = c.getUnboundDeclaration() |
-    result.hasBody() and
+    (
+      result.hasBody()
+      or
+      // take synthesized bodies into account, e.g. implicit constructors
+      // with field initializer assignments
+      result = any(ControlFlow::Nodes::ElementNode n).getEnclosingCallable()
```

Comment:
Should we add a `not result.hasBody()` and maybe a type check `result instanceof InstanceConstructor`  to make it explicit what case we cover here?

Reply:
I would rather not, actually, as we are then also prepared for cases where we might synthesize other kinds of bodies in the future.